test/components/legendTests.ts

```diff
@@ -350,3 +350,183 @@ describe("Legend", () => {
+    assert.operator(upperLabelBCR.left, "<=", swatchContainerBCR.left, "second label to left of swatches");
+    assert.operator(upperLabelBCR.bottom, "<=", lowerLabelBCR.top, "lower label is drawn below upper label");
+
+    svg.remove();
+  });
+
+  it("re-renders when scale domain updates", () => {
+    var legend = new Plottable.Component.InterpolatedColorLegend(colorScale, "horizontal");
+    legend.renderTo(svg);
+
+    colorScale.domain([0, 85]);
+    assertBasicRendering(legend);
+
+    svg.remove();
+  });
+
+  it("orient() input-checking", () => {
+    var legend = new Plottable.Component.InterpolatedColorLegend(colorScale, "horizontal");
+
+    legend.orient("horizontal"); // should work
+    legend.orient("right"); // should work
+    legend.orient("left"); // should work
+
+    assert.throws(() => legend.orient("blargh"), "not a valid orientation");
+    svg.remove();
+  });
+
+  it("orient() triggers layout computation", () => {
+    var legend = new Plottable.Component.InterpolatedColorLegend(colorScale, "horizontal");
+    legend.renderTo(svg);
+
+    var widthBefore = legend.width();
+    var heightBefore = legend.height();
+
+    legend.orient("right");
+    assert.notEqual(legend.width(), widthBefore, "proportions changed (width)");
+    assert.notEqual(legend.height(), heightBefore, "proportions changed (height)");
+    svg.remove();
+  });
+
+  it("renders correctly when width is constrained (orientation: horizontal)", () => {
+    svg.attr("width", 100);
+    var legend = new Plottable.Component.InterpolatedColorLegend(colorScale, "horizontal");
+    legend.renderTo(svg);
+    assertBasicRendering(legend);
+    svg.remove();
+  });
+
+  it("renders correctly when height is constrained (orientation: horizontal)", () => {
+    svg.attr("height", 20);
+    var legend = new Plottable.Component.InterpolatedColorLegend(colorScale, "horizontal");
+    legend.renderTo(svg);
+    assertBasicRendering(legend);
+    svg.remove();
+  });
+
+  it("renders correctly when width is constrained (orientation: right)", () => {
+    svg.attr("width", 30);
+    var legend = new Plottable.Component.InterpolatedColorLegend(colorScale, "right");
+    legend.renderTo(svg);
+    assertBasicRendering(legend);
+    svg.remove();
+  });
+
+  it("renders correctly when height is constrained (orientation: right)", () => {
+    svg.attr("height", 100);
+    var legend = new Plottable.Component.InterpolatedColorLegend(colorScale, "right");
+    legend.renderTo(svg);
+    assertBasicRendering(legend);
+    svg.remove();
+  });
+
+  it("renders correctly when width is constrained (orientation: left)", () => {
+    svg.attr("width", 30);
+    var legend = new Plottable.Component.InterpolatedColorLegend(colorScale, "left");
+    legend.renderTo(svg);
+    assertBasicRendering(legend);
+    svg.remove();
+  });
+
+  it("renders correctly when height is constrained (orientation: left)", () => {
```

Comment:
I think the core of my problem is mainly from the term "renders correctly"

While we do this in a bunch of places, I think the term is way too imprecise for what we are really looking for.  Figuring this out could take some time though, so I'm just going to focus on reviewing the mother assertBasicRendering function while thinking about a better way to structure these tests.